src/pages/register/register.tsx

```diff
@@ -1,7 +1,15 @@
 import React from "react";
-import { RegisterModal } from "../../components/user/registerModal";
+import { RegisterModal } from "../../components/user/";
+import { useTokenContext } from "../../context";
+import { redirectToHome } from "../../utils";
 
 export const RegisterPage: React.FC = () =>{
+    const {token} = useTokenContext();
+    
+    if(token !== null){

```

Comment:
as you can see that we repeat this flow in almost every page. is there any way to make this part elegant?